docs/source/_notebooks/keras-image-classifiers.rst

```diff
@@ -0,0 +1,435 @@
+
+Explaining Keras image classifier predictions
+=============================================
+
+If we have a model that takes in an image as its input, and outputs
+class scores, i.e. probabilities that a certain object is present in the
+image, then we can use ELI5 to check what is it in the image that made
+the model predict a certain class score. We do that using a method
+called 'Grad-CAM'.
+
+We will be using images from ImageNet (for more information see
+http://image-net.org/), and classifiers from ``keras.applications``.
+
+This has been tested with Python 3.7.3, Keras 2.2.4, and Tensorflow
+1.13.1.
+
+1. Explaining the model's prediction
+------------------------------------
+
+To start out, lets load our image classifier and data.
+
+.. code:: ipython3
+
+    from PIL import Image
+    from IPython.display import display
+    
+    import logging
+    import tensorflow as tf
+    tf.get_logger().setLevel(logging.ERROR) # disable Tensorflow warnings for this tutorial
+    import warnings
+    warnings.simplefilter("ignore") # disable Keras warnings for this tutorial
+    import keras
+    from keras.applications import mobilenet_v2
+    
+    import eli5
+    
+    # let's instantiate our model
+    model = mobilenet_v2.MobileNetV2(include_top=True, weights='imagenet', classes=1000)
+    
+    # and load our sample image
+    image = 'imagenet-samples/cat_dog.jpg'  # TODO: replace this image with something more interesting
+    
+    dims = model.input_shape[1:3] # 224x224
+    doc = eli5.keras.image_from_path(image, image_shape=dims)
```

Comment:
I think this is not the best way to load an image for the purpose of tutorial, because the user already knows how to load images, and might have images already loaded, but it's not clear from this what is ``doc`` and how to convert an already loaded image to this. I would suggest  to load an image in some way which would be recognized by the user (e.g. Image.open or keras.preprocessing.image.load_img), show the type and dimensions of resulting image, then apply any post-processing if needed, and again show result.

Reply:
Indeed using functions recognisable by the user is clearer. Should I remove the `eli5.keras.image_from_path()` function from the API? May be too specialised.

Reply:
Also should I remove the `eli5.keras.image_from_path()` function from the API? May be too specialised.

Reply:
> Also should I remove the eli5.keras.image_from_path() function from the API? May be too specialised.

Yes, I'd rather remove it from the API

Reply:
https://github.com/TeamHG-Memex/eli5/pull/315/commits/a923864ae355b5597ec398a33b9cd9f43f2c4250